src/stan/lang/function_signatures.h

```diff
@@ -2,39 +2,39 @@
 // src/stan/lang/ast.hpp
 
 std::vector<base_expr_type> base_types;
-base_types.push_back(INT_T);
-base_types.push_back(DOUBLE_T);
-base_types.push_back(VECTOR_T);
-base_types.push_back(ROW_VECTOR_T);
-base_types.push_back(MATRIX_T);
+base_types.push_back(int_type());
+base_types.push_back(double_type());
+base_types.push_back(vector_type());
+base_types.push_back(row_vector_type());
+base_types.push_back(matrix_type());
 
 std::vector<expr_type> vector_types;
-vector_types.push_back(DOUBLE_T);  // scalar
-vector_types.push_back(expr_type(DOUBLE_T, 1U));  // std vector
-vector_types.push_back(VECTOR_T);  // Eigen vector
-vector_types.push_back(ROW_VECTOR_T);  // Eigen row vector
+vector_types.push_back(expr_type(double_type()));  // scalar
+vector_types.push_back(expr_type(double_type(), 1U));  // std vector
+vector_types.push_back(expr_type(vector_type()));  // Eigen vector
```

Comment:
You can define local variables or global constants for all these things that get reused all over the place.  No big deal leaving it this way.  But I'm not seeing any implicit promotions, so you can probalby make `expr_type` constructors `explicit`.